### ISSUE TITLE
Prefix NuShell environment variables

### ISSUE THREAD

**Issue by samuel-lindgren** [nushell/nushell#9636]
### Related problem

I think it is generally good practise to prefix environment variables to ensure that they do not collide with environment varibles from other programs. One such case is `LOG_FORMAT` which seems to be set [here](https://github.com/nushell/nushell/blob/667502e8da0ef8bc06c10f62684c76debae4cf50/crates/nu-std/std/log.nu#L34). This environment variable in particular cause issues for me, since we use LOG_FORMAT in my current project. However, I am encouraging to prefix it there as well.

### Describe the solution you'd like

My suggestion would be to prefix the NuShell environment variables, with something like `NUSHELL_LOG_FORMAT`.

**Comment by sholderbach** [comment]
Good point! Although solving problems with longer stringly typed symbols is not really in the spirit of the structured approach of Nushell, we should be more consistent and get our story straight.

The particular offender in your example is part of the std library logging system contributing:
```
LOG_ANSI
LOG_FORMAT
LOG_LEVEL
LOG_PREFIX
LOG_SHORT_PREFIX
```

The special `$env.config` is a nested structure.

We also have defined as `$env` variables partly for historical reasons and to be includable in `env.nu`:

```
ENV_CONVERSIONS
PROMPT_COMMAND
PROMPT_COMMAND_RIGHT
PROMPT_INDICATOR
PROMPT_INDICATOR_VI_INSERT
PROMPT_INDICATOR_VI_NORMAL
PROMPT_MULTILINE_INDICATOR
```

As an example for `$env` vars with a prefix we have:

```
NU_LIB_DIRS
NU_PLUGIN_DIRS
```

**Comment by sholderbach** [comment]
Related issue mentioning pollution of the global `$env` scope: #9276
